File: Quorum3Plugins/GamePlugin/JavaGame/src/plugins/quorum/Libraries/Game/Graphics/Material.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package plugins.quorum.Libraries.Game.Graphics;

/**
 *
 * @author Bill
 */
public class Material 
{
    public java.lang.Object me_ = null;
    
    private static int counter = 0;
    
    public String GenerateDefaultID()
    {
        return "mtl" + (++counter);
    }
}

File: Quorum3/Library/Standard/Plugins/javascript/Libraries/Game/Graphics/Material.js
function plugins_quorum_Libraries_Game_Graphics_Material_() 
{
    this.GenerateDefaultID = function() 
    {
        if (!plugins_quorum_Libraries_Game_Graphics_Material_.counter)
        {
            plugins_quorum_Libraries_Game_Graphics_Material_.counter = 1;
            return 1;
        }
        
        plugins_quorum_Libraries_Game_Graphics_Material_.counter++;
        return plugins_quorum_Libraries_Game_Graphics_Material_.counter;
    };
}
